src/kontrol/prove.py

```diff
@@ -419,8 +419,9 @@ def _method_to_cfg(
         new_node_ids = [init_node.id]
         init_node_id = init_node.id
 
-    is_test = method.signature.startswith('test')
-    failing = method.signature.startswith('testFail')
+    proof_prefixes = ['test', 'prove', 'check']
+    is_test = any(method.signature.startswith(prefix) for prefix in proof_prefixes)
+    failing = any(method.signature.startswith(prefix + 'Fail') for prefix in proof_prefixes)
```

Comment:
The logic for deciding what sort of KCFG to construct based on the function name should be pulled up from this function, and `is_test` and `failing` should become parameters (not necessarily in this PR though).

Additionally, `proof_prefixes` is the sort of thing that should be configurable through a configuration file. If it doesn't break Foundry we can consider adding a section to `foundry.toml`, otherwise maybe its time to introduce `kontrol.toml`. This would enable cleaning up the CLI a bit as well.

Reply:
> The logic for deciding what sort of KCFG to construct based on the function name should be pulled up from this function

Then this feature could also be tested with a simple unit test.

Reply:
Yes, thanks Tamás, I agree! Halmos allows specifying the function name using the following argument, and we can do the same thing, introducing, e.g., `--function` and `--fail-function` as the first step:
```
    parser.add_argument(
        "--function",
        metavar="FUNCTION_NAME_PREFIX",
        default="check_",
        help="run tests matching the given prefix. Shortcut for `--match-test '^{PREFIX}'`. (default: '%(default)s')",
    )
```

And agreed on `.toml` configuration, we have in on a roadmap for February, but I wouldn't mind doing it earlier if I have time next week. Adding a new section (e.g., `[kontrol]` or `[symbolic]`) won't break Foundry but it will throw a warning:
```
warning: Found unknown config section in foundry.toml: [kontrol]
This notation for profiles has been deprecated and may result in the profile not being registered in future versions.
Please use [profile.kontrol] instead or run `forge config --fix`.
```

Reply:
@JuanCoRo needs this for an engagement, so I'm leaning towards leaving this PR as it right now, as it's the format we've agreed on so far, but I'll open a new PR right after that, if that works. Or please let me know what you think @JuanCoRo!